posthog/api/plugin.py

```diff
@@ -0,0 +1,132 @@
+        if not settings.PLUGINS_INSTALL_FROM_WEB:
+            return Response([])
+        url = "https://raw.githubusercontent.com/PostHog/plugins/main/repository.json"
+        plugins = requests.get(url)
+        return Response(json.loads(plugins.text))
+
+    def destroy(self, request: request.Request, pk=None) -> Response:  # type: ignore
+        if not settings.PLUGINS_INSTALL_FROM_WEB:
+            raise APIException("Plugin installation via the web is disabled!")
+        plugin = Plugin.objects.get(pk=pk)
+        if plugin.from_json:
+            raise APIException('Can not delete plugin "{}", which is configured from posthog.json!'.format(plugin.name))
+        plugin.delete()
+        reload_plugins_on_workers()
+        return Response(status=204)
+
+
+class PluginConfigSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = PluginConfig
+        fields = ["id", "plugin", "enabled", "order", "config"]
+
+    def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> PluginConfig:
+        if not settings.PLUGINS_CONFIGURE_FROM_WEB:
+            raise APIException("Plugin configuration via the web is disabled!")
+        request = self.context["request"]
+        plugin_config = PluginConfig.objects.create(team=request.user.team, **validated_data)
+        reload_plugins_on_workers()
+        return plugin_config
+
+    def update(self, plugin_config: PluginConfig, validated_data: Dict, *args: Any, **kwargs: Any) -> PluginConfig:  # type: ignore
+        if not settings.PLUGINS_CONFIGURE_FROM_WEB:
+            raise APIException("Plugin configuration via the web is disabled!")
+        plugin_config.enabled = validated_data.get("enabled", plugin_config.enabled)
+        plugin_config.config = validated_data.get("config", plugin_config.config)
+        plugin_config.order = validated_data.get("order", plugin_config.order)
+        plugin_config.save()
+        reload_plugins_on_workers()
+        return plugin_config
+
+
+class PluginConfigViewSet(viewsets.ModelViewSet):
+    queryset = PluginConfig.objects.all()
+    serializer_class = PluginConfigSerializer
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        if not settings.PLUGINS_CONFIGURE_FROM_WEB:
+            return queryset.none()
+        return queryset.filter(team_id=self.request.user.team.pk)
+
+    # we don't use this endpoint, but have something anyway to prevent team leakage
+    def destroy(self, request: request.Request, pk=None) -> Response:  # type: ignore
+        if not settings.PLUGINS_CONFIGURE_FROM_WEB:
+            raise APIException("Plugin configuration via the web is disabled!")
+        plugin_config = PluginConfig.objects.get(team=request.user.team, pk=pk)
+        plugin_config.enabled = False
+        plugin_config.save()
+        return Response(status=204)
+
+    @action(methods=["GET"], detail=False)
+    def global_plugins(self, request: request.Request):
```

Comment:
Maybe it'd be a bit more RESTful to include the global plugins by default on the main GET response and then just allowing a QS filter to show only global plugins?